lib/Service/ImageService.php

```diff
@@ -0,0 +1,596 @@
+use OCP\IPreview;
+use OCP\Share\Exceptions\ShareNotFound;
+use OCP\Share\IShare;
+use Throwable;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\ServerException;
+use OCA\Text\AppInfo\Application;
+use OCP\Http\Client\IClientService;
+use OCP\Files\IRootFolder;
+use Psr\Log\LoggerInterface;
+use OCP\Share\IManager as ShareManager;
+use function json_encode;
+use function preg_replace;
+
+class ImageService {
+
+	/**
+	 * @var ShareManager
+	 */
+	private $shareManager;
+	/**
+	 * @var IRootFolder
+	 */
+	private $rootFolder;
+	/**
+	 * @var IClientService
+	 */
+	private $clientService;
+	/**
+	 * @var LoggerInterface
+	 */
+	private $logger;
+	/**
+	 * @var IPreview
+	 */
+	private $previewManager;
+
+	public function __construct(IRootFolder $rootFolder,
+								LoggerInterface $logger,
+								ShareManager $shareManager,
+								IPreview $previewManager,
+								IClientService $clientService) {
+		$this->rootFolder = $rootFolder;
+		$this->shareManager = $shareManager;
+		$this->clientService = $clientService;
+		$this->logger = $logger;
+		$this->previewManager = $previewManager;
+	}
+
+	/**
+	 * Get image content from file name
+	 *
+	 * @param int $textFileId
+	 * @param string $imageFileName
+	 * @param string $userId
+	 * @return File|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 * @throws \OCP\Lock\LockedException
+	 * @throws \OC\User\NoUserException
+	 */
+	public function getImage(int $textFileId, string $imageFileName, string $userId): ?ISimpleFile {
+		$textFile = $this->getTextFile($textFileId, $userId);
+		$attachmentFolder = $this->getOrCreateAttachmentDirectoryForFile($textFile);
+		if ($attachmentFolder !== null) {
+			try {
+				$imageFile = $attachmentFolder->get($imageFileName);
+			} catch (NotFoundException $e) {
+				return null;
+			}
+			if ($imageFile instanceof File) {
+//				return $imageFile;
+				return $this->previewManager->getPreview($imageFile, 1024, 1024);
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Get image content from file name in public context
+	 *
+	 * @param int $textFileId
+	 * @param string $imageFileName
+	 * @param string $shareToken
+	 * @return File|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 * @throws \OCP\Lock\LockedException
+	 * @throws \OC\User\NoUserException
+	 */
+	public function getImagePublic(int $textFileId, string $imageFileName, string $shareToken): ?ISimpleFile {
+		$textFile = $this->getTextFilePublic($textFileId, $shareToken);
+		$attachmentFolder = $this->getOrCreateAttachmentDirectoryForFile($textFile);
+		if ($attachmentFolder !== null) {
+			try {
+				$imageFile = $attachmentFolder->get($imageFileName);
+			} catch (NotFoundException $e) {
+				return null;
+			}
+			if ($imageFile instanceof File) {
+//				return $imageFile;
```

Comment:
- [ ] To be discussed if we still need the original file for gif support for example